File: leetcode/src/main/java/array/CheckIfAWordOccursAsAPrefixOfAnyWordInASentence.java
package array;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

/*
    https://leetcode.com/problems/check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence/

    Example 1:

        Input: sentence = "i love eating burger", searchWord = "burg"
        Output: 4
        Explanation: "burg" is prefix of "burger" which is the 4th word in the sentence.
 */
public class CheckIfAWordOccursAsAPrefixOfAnyWordInASentence {

    public int isPrefixOfWord(String sentence, String searchWord) {
        String[] arr = sentence.split(" ");
        for (int i = 0; i < arr.length; i++) {
            if (arr[i].startsWith(searchWord)) {
                return i + 1;
            }
        }
        return -1;
    }

    @Test
    public void test() {
        assertEquals(4, isPrefixOfWord("i love eating burger", "burg"));
        assertEquals(2, isPrefixOfWord("this problem is an easy problem", "pro"));
        assertEquals(-1, isPrefixOfWord("i am tired", "you"));
    }

}

File: py-algorithm/arrays/check_if_a_word_occurs_as_a_prefix_of_any_word_in_a_sentence.py
import unittest

"""
    https://leetcode.com/problems/check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence/
"""


class Solution(unittest.TestCase):

    def isPrefixOfWord(self, sentence: str, searchWord: str) -> int:
        for i, word in enumerate(sentence.split(" ")):
            if word.startswith(searchWord):
                return i + 1

        return -1

    def test(self):
        self.assertEqual(4, self.isPrefixOfWord("i love eating burger", "burg"))


if __name__ == '__main__':
    unittest.main()
